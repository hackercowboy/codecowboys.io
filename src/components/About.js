import React from 'react';
import PropTypes from 'prop-types';
import { Parallax } from 'react-parallax';
import { injectIntl } from 'react-intl';

import Section from './Section';

import './About.scss';

const aboutIcon = require('../assets/images/david.png');
const aboutBackground = require('../assets/images/bg-5.jpg');

function About({ intl }) {
  return (
    <Parallax
      bgImage={aboutBackground}
      strength={500}
    >
      <Section
        id="about"
        title={intl.formatMessage({ id: 'about.title' })}
        subtitle={intl.formatMessage({ id: 'about.subtitle' })}
        dark
      >
        <h4>David Ãœbelacker</h4>
        <p>{intl.formatMessage({ id: 'about.content_1' })}</p>
        <p>{intl.formatMessage({ id: 'about.content_2' })}</p>
        <img
          alt={intl.formatMessage({ id: 'about.photo_alt' })}
          className="rounded-circle"
          src={aboutIcon}
          style={{ width: '100px', height: '100px' }}
        />
      </Section>
    </Parallax>
  );
}

About.propTypes = {
  intl: PropTypes.shape({ formatMessage: PropTypes.func.isRequired }).isRequired,
};

export default injectIntl(About);
