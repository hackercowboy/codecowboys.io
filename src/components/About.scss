.about {
  margin-bottom: 50px;

  h4 {
    margin: 0;
  }
}

.about-wrapper {
  align-items: center;
  display: flex;
}

.about-teaser {
  flex-grow: 1;
}

.about-image {
  margin-left: 10px;
  width: 100px;

  img {
    border-radius: 50px;
  }
}
