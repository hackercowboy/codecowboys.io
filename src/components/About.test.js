import React from 'react';
import { shallow } from 'enzyme';

import About from './About';

jest.mock('react-intl');

describe('<About/>', () => {
  it('should render', () => {
    const wrapper = shallow(<About />);
    expect(wrapper.shallow().debug()).toMatchSnapshot();
  });
});
