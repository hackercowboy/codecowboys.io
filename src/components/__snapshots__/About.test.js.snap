// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About/> should render 1`] = `
"<Parallax bgImage=\\"bg-5.jpg\\" strength={500} bgClassName=\\"react-parallax-bgimage\\" bgImageAlt=\\"\\" className=\\"\\" contentClassName=\\"\\" disabled={false}>
  <Section id=\\"about\\" title=\\"About\\" subtitle=\\"I code great apps like a cowboy!\\" dark={true} light={false} teaser={[undefined]} big={false}>
    <div>
      <div body={true}>
        <h4>
          David Übelacker
        </h4>
        <p>
          I am a passionate software engineer working for a Swiss company.
        </p>
        <p>
          From time to time i work on some nice projects in my free time, which i publish on this site.
        </p>
      </div>
      <div right={true} className=\\"ml-3\\">
        <img alt=\\"Photo of David Übelacker\\" className=\\"rounded-circle\\" src=\\"david.png\\" style={{...}} />
      </div>
    </div>
  </Section>
</Parallax>"
`;
